hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/ReactiveResultSetProcessor.java

```diff
@@ -0,0 +1,91 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.loader;
+
+import org.hibernate.AssertionFailure;
+import org.hibernate.engine.internal.TwoPhaseLoad;
+import org.hibernate.engine.spi.*;
+import org.hibernate.event.spi.PreLoadEvent;
+import org.hibernate.event.spi.PreLoadEventListener;
+import org.hibernate.internal.CoreLogging;
+import org.hibernate.internal.CoreMessageLogger;
+import org.hibernate.loader.plan.exec.query.spi.NamedParameterContext;
+import org.hibernate.loader.spi.AfterLoadAction;
+import org.hibernate.reactive.session.ReactiveSession;
+import org.hibernate.reactive.util.impl.CompletionStages;
+import org.hibernate.transform.ResultTransformer;
+
+import java.io.Serializable;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+public interface ReactiveResultSetProcessor {
```

Comment:
Is this supposed to be a reactive version of `ResultSetProcessor` in core or does it have a different purpose?

Reply:
hibernate-orm has a `ResultSetProcessor` that is used by `LoadPlan`-based loaders. 

I added `ReactiveResultSetProcessor` to unify how `ReactiveLoader#doReactiveQueryAndInitializeNonLazyCollections` processes a `ResultSet`s  by  both `Loader`-based and `LoadPlan`-based loaders.

Reply:
OK fine, so it’s something new.